libsolidity/parsing/Parser.cpp

```diff
@@ -1614,9 +1614,18 @@ ASTPointer<Expression> Parser::parsePrimaryExpression()
 		}
 		break;
 	case Token::StringLiteral:
+	{
+		string literal = m_scanner->currentLiteral();
+		while (m_scanner->peekNextToken() == Token::StringLiteral)
```

Comment:
I think this whole change could be simplified and shortened by introducing `Token::HexStringLiteral` and using this loop here to concatenate string literals and hex string literals while checking that their type stays the same. This would also allow us to create a better error message.

Reply:
I've introduced `Token::HexStringLiteral` here af7d0028fbd7345865f3eca87f36ba5c973ff42a and made the relevant changes.